index.rst

```diff
@@ -1,61 +1,311 @@
+
+The command would simply receive an array of wavefront errors and apply them to the system properly registering the correction internally.
+
+.. code-block:: py
+
+  import numpy as np
+
+  # Assuming the wavefront correction is an array of 19 Zernike coefficients
+  wf = np.zeros(19)
+
+  # Assuming ANSI indexes, wf[7] is vertical comma
+  # Adding 1mm of vertical comma
+  wf[7] = 1.
+
+  await mtaos.cmd_runOFC.set_start(wf = wf)
+
+When the ``MTAOS`` is configured it will load appropriate configuration files for the OFC.
+During commissioning it is likely we will want to run the ``OFC`` with a different set of configurations.
+In order to allow quick turnaround, the command can accept an additional yaml string with overriding  parameters for the currently loaded configuration.
+
+For instance, one could truncate the sensitivity matrix so that a wavefront error correction is applied to a specific component (or components), e.g.;
+
+.. code-block:: py
+
+  import yaml
+
+  # switch off all corrections except m2 hexapod
+  # See https://github.com/lsst-ts/ts_ofc/blob/master/policy/zkAndDofIdxArraySet.yaml
+  ofc_config = {
+    "dofIdx": {
+        "m2Hex": [1, 1, 1, 1, 1],
+        "camHex": [0, 0, 0, 0, 0],
+        "m1m3Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        "m2Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      }
+  }
+
+  # convert dictionary to yaml string
+  config = yaml.safe_dump(ofc_config)
+
+  await mtaos.cmd_runOFC.set_start(wf = wf, config = config)
+
+Similarly, the user can manipulate the configuration for Camera hexapod, M2 and M1M3.
+Furthermore, this gives users the capability of customizing the configuration of the OFC on the fly, without the need to recycle the CSC state and the need for several different configurations.
+By relying on a yaml string to parse the configuration it is also possible to expand the feature as needed without the need to update the xml interface, which considerably simplify the redeployment cycle.
+
+.. _Architecture-updates:
+
+Architecture
+============
+
+In terms of the MTAOS itself, there is practically no substantial architecture changes needed (see :numref:`fig-mtaosClass` bellow).
+The current code-base is well organized in the sense that the "heavy lifting" is encapsulated by the `Model class`_.
+
+.. figure:: /_static/mtaosClass.png
+   :name: fig-mtaosClass
+   :target: ../_images/mtaosClass.png
+   :alt: MTAOS class diagram
+
+   MTAOS class diagram (`original uml file from v0.4.5`_).
+
+
+.. _Model class: https://ts-mtaos.lsst.io/py-api/lsst.ts.MTAOS.Model.html#lsst.ts.MTAOS.Model
+.. _original uml file from v0.4.5: https://github.com/lsst-ts/ts_MTAOS/blob/v0.4.5/doc/uml/mtaosClass.uml
+
+Nevertheless, the `Model class`_ itself will require substantial updates to incorporate the new functionality mentioned :ref:`above <Data-access-changes>` and the conversion of `WEP`_ into a gen3 pipeline task.
```

Comment:
The `Model` class depends on the interface classes in `ts_wep/python/lsst/ts/wep/ctrlIntf` and `ts_ofc/python/lsst/ts/ofc/ctrlIntf/`. Therefore, if these two class groups have no change, there should be no problem for the `Model` class. This means the use of gen3 should not affect the `Model` class here. Instead, I think only the interface classes in `ts_wep` and `ts_ofc` need to be updated to use the gen3 pipeline task.